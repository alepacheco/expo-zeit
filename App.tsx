import React, { useState } from 'react';
import { StyleSheet, Text, View, Image } from 'react-native';
import axios from 'axios';
import { API_URL } from 'react-native-dotenv';

const App = () => {
  const [message, setMessage] = useState('Loading');

  (async () => {
    try {
      const { data } = await axios.get(API_URL + '/');
      setMessage(data);
    } catch (error) {
      setMessage('Api not workign.');
    }
  })();

  return (
    <View style={styles.container}>
      <Text>Zeit now & expo (React Native) Template!</Text>
      <Text>Published to Expo and Next</Text>

      <Text
        style={{
          backgroundColor: 'aliceblue',
          padding: 14,
          margin: 24,
          borderRadius: 6,
        }}>
        {message}
      </Text>
      <Image
        style={{ width: 50, height: 50 }}
        source={{
          uri: 'https://facebook.github.io/react-native/img/tiny_logo.png',
        }}
      />
    </View>
  );
};

export default App;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
