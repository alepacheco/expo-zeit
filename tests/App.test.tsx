import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';
jest.mock('axios');

describe('<App />', () => {
  it('renders correctly', async () => {
    let tree: null | renderer.ReactTestRenderer = null;

    await renderer.act(async () => {
      tree = renderer.create(<App />);
    });
    expect(tree.toJSON()!.children!.length).toBe(4);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
