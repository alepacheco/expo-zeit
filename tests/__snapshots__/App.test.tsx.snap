// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Zeit now & expo (React Native) Template!
  </Text>
  <Text>
    Published to Expo and Next
  </Text>
  <Text
    style={
      Object {
        "backgroundColor": "aliceblue",
        "borderRadius": 6,
        "margin": 24,
        "padding": 14,
      }
    }
  >
    Api not workign.
  </Text>
  <Image
    source={
      Object {
        "uri": "https://facebook.github.io/react-native/img/tiny_logo.png",
      }
    }
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
</View>
`;
